sycl/include/CL/sycl/handler.hpp

```diff
@@ -1351,6 +1351,12 @@ class __SYCL_EXPORT handler {
   ///
   /// \param Event is a valid SYCL event to wait on.
   void depends_on(event Event) {
+    if (info::event_command_status::ext_oneapi_unknown ==
+        Event.get_info<info::event::command_execution_status>()) {
```

Comment:
This looks very expensive. Since what we really would like to know here is whether the event is discarded, can we just check this field?

Reply:
This code covers the case "when invalid event is passed into handler API" described in https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/DiscardQueueEvents/SYCL_EXT_ONEAPI_DISCARD_QUEUE_EVENTS.asciidoc

Here we just want to check if the `MState` field of `event_impl` has `HES_Invalid` value or not. 
I think it is possible if we move the definition of the `depends_on()` method to the `handler.cpp` and check the field by the new public method of `event_impl`(it needs to be added). I will try that, thanks!